jwt-bearer-auth-yesod/src/Network/Wai/Lens.hs

```diff
@@ -0,0 +1,74 @@
+-- |
+-- This module is incomplete, but provides a couple of lenses to inspect a
+-- WAI request.
+module Network.Wai.Lens
+  ( requestMethodL
+  , requestHeadersL
+  , atHeaderName
+  , view
+  , set
+  , over
+  ) where
+
+import Prelude
+
+import qualified Data.ByteString as BS
+import Data.Functor.Const
+import Data.Functor.Identity
+import qualified Network.HTTP.Types as H
+import Network.Wai
+
+requestMethodL
+  :: forall f
+   . Functor f
+  => (H.Method -> f H.Method)
+  -> Request
+  -> f Request
+requestMethodL f req =
+  let fMethod = f (requestMethod req)
+  in  fmap (\m -> req {requestMethod = m}) fMethod
```

Comment:
I rarely see lens built "by hand" like this, I usually do

```suggestion
requestMethodL = lens requestMethod $ \x y -> x { requestMethod = y }
```

Did you do this so as not to concretely depend on `lens`? I get that, but I'll usually bring in `microlens` instead so it can be just as nice without the huge dependency.

Reply:
Did you want to simplify this to use `lens`? It seems you agreed the dep is already in use, so IMO there's no reason to do it this way.

Reply:
Whatever you want, it just wasn't clear to me.

Reply:
Next pr already open as draft 😁